src/Futurice/Time/Month.hs

```diff
@@ -130,6 +130,22 @@ instance Enum Month where
         let (y, m) = divMod i 12
         in Month (fromIntegral y) (toEnum $ m + 1)
 
+instance Csv.ToField Month where
+    toField = Csv.toField . monthToString
+
+-- | TODO: safer parsing method for this that doesn't use read for the month parts
+instance Csv.FromField Month where
+    parseField field | BS.length field == 7 = parsed
+                     | otherwise            = empty
+        where
+            parsed =
+                case map BS.unpack $ BS.split '-' field of
+                    []          -> empty
+                    [yr, mo]    -> pure $ Month (read yr :: Integer) (toMonth mo)
+                    _           -> empty
+
+            toMonth = toEnum . read
+
```

Comment:
I would probably try something like this though this is pretty ugly solution also with needing to convert to text first
```haskell
instance Csv.FromField Month where
    parseField field = let monthtext = TE.decodeUtf8With TE.lenientDecode field
                           month = parseUrlPiece monthtext :: Either Text Month
                       in case month of
                            Left _ -> empty
                            Right m -> pure m
```

Reply:
You can also use ` AT.parseOnly (mkMonth <$> Parsers.month)` like thing, just change from `Data.Attoparsec.Text` to `Data.Attoparsec.ByteString`.